packages/alfa-rules/src/common/predicate/is-offscreen.ts

```diff
@@ -0,0 +1,46 @@
+import { Cache } from "@siteimprove/alfa-cache";
+import { Device } from "@siteimprove/alfa-device";
+import { Element, Node } from "@siteimprove/alfa-dom";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Context } from "@siteimprove/alfa-selector";
+import { Style } from "@siteimprove/alfa-style";
+
+const { abs } = Math;
+const { isElement } = Element;
+
+const cache = Cache.empty<Device, Cache<Context, Cache<Node, boolean>>>();
+
+export function isOffscreen(
+  device: Device,
+  context: Context = Context.empty()
+): Predicate<Node> {
+  return function isOffscreen(node) {
+    return cache
+      .get(device, Cache.empty)
+      .get(context, Cache.empty)
+      .get(node, () => {
+        if (isElement(node)) {
+          const style = Style.from(node, device, context);
+
+          const { value: position } = style.computed("position");
+          const { value: left } = style.computed("left");
+          const { value: right } = style.computed("right");
+
+          if (position.value !== "static") {
+            for (const inset of [left, right]) {
+              switch (inset.type) {
+                case "length":
+                  return abs(inset.value) >= 9999;
```

Comment:
Also, obvious prone to errors on huge displays, or with huge elements… Not sure we can do much better without a layout system.
(also, ACT rules is having `-999em` in at least two cases, which we won't catch here)

Reply:
These are computed `px` values so `em` values will have been resolved by the time this condition is reached. `-999em` can be expected to compute to at least `-9999px` for reasonable font sizes; all it takes is `10px`, after all.